rdmo/core/xml.py

```diff
@@ -1,38 +1,125 @@
 import logging
 import re
+from collections import OrderedDict
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import Optional
+
+from django.utils.translation import gettext_lazy as _
 
 import defusedxml.ElementTree as ET
-from packaging.version import parse
-
-log = logging.getLogger(__name__)
-
-models = {
-  'catalog': 'questions.catalog',
-  'section': 'questions.section',
-  'page': 'questions.page',
-  'questionset': 'questions.questionset',
-  'question': 'questions.question',
-  'attribute': 'domain.attribute',
-  'optionset': 'options.optionset',
-  'option': 'options.option',
-  'condition': 'conditions.condition',
-  'task': 'tasks.task',
-  'view': 'views.view'
-}
-
-
-def read_xml_file(file_name):
+from packaging.version import Version, parse
+
+from rdmo import __version__ as VERSION
+from rdmo.core.constants import RDMO_MODELS
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_RDMO_XML_VERSION = '1.11.0'
+ELEMENTS_USING_KEY = {RDMO_MODELS['attribute']}
+
+
+@dataclass
+class XmlToElementsParser:
```

Comment:
Why is this a data class and not a regular class?

Reply:
yes, you're right this was not ideal.
I've reverted and refactored this back into the functional approach.
Each function returns something like `obj, error` to keep the error handling consistent.